torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -957,6 +957,10 @@ void Reducer::mark_bucket_ready(size_t bucket_index) {
   }
 }
 
+void Reducer::install_futures(c10::List<c10::intrusive_ptr<c10::ivalue::Future>> futs) {
+    installed_futures_ = std::move(futs);
```

Comment:
could we append futs here instead of moving?  it is possible that install_futures() was called multiple times and we need to wait all these installed futures?

Reply:
Great idea!